Add Data-Wrangler sub-chapter (#2883)
alan-turing-institute/the-turing-way: book/website/collaboration/research-infrastructure-roles/data-wrangler.md

(cl-infrastructure-data-wranglers-tasks)=
## What do Data Wranglers do?

In a data science project, it is commonly observed that data wrangling tasks take the majority of time {cite:ps}`anaconda2020sods`, in contrast to data analysis and modelling.
Traditionally, data wrangling tasks involve cleaning, restructuring and filtering data into standardised research ready formats.
However, in terms of Data Wrangler as a profession, day to day tasks and objectives can be much more diverse.   

Data Wranglers work primarily in the space between data generators and data analysts, who are addressing the research question of interest.
Understanding the intended use for the data in the context of the analysis and research questions, Data Wranglers can be in the position of influencing data generators in improving data collection methods, ultimately benefiting all points of the data life cycle.
Similarly, Data Wranglers will conduct preliminary analysis on the data to ensure both completeness of data and preparation for data analysis, acting as a proxy for the data generator’s knowledge during the data analysis process.

> As a key focus of a Data Wrangler’s role is the preparation of research ready data {cite:ps}`stewart2022table`, data security, data management and FAIR standards are core to the role {cite:ps}`mons2017fair`.

Reviewer: This line reads a little confusing to me. Perhaps:
> A key focus of a Data Wrangler’s role is the preparation of research ready data {cite:ps}`stewart2022table` in which data security, data management and FAIR standards are also core responsibilities {cite:ps}`mons2017fair`

Author: Much better, thanks! But maybe "responsibilities" can be replaced by "priorities" and then it sounds even better?

Author: > A key focus of a Data Wrangler’s role is the preparation of research ready data {cite:ps}stewart2022table in which data security, data management and FAIR standards are all core priorities {cite:ps}mons2017fair

Reviewer: purrfect 🐈

### Examples of day-to-day tasks:

* Data retrieval, querying, data analysis and visualizations
* Data pipeline development, maintenance and improvement
* Communicating with domain and technical experts
* Data privacy and infrastructure setup and management
* Summarising domain specific research papers
* Sharing and reviewing resources and code
* Maintaining communication across multiple teams
* Running workshops on data wrangling and related skills

(cl-infrastructure-data-wranglers-skills)=
## What qualifications or skills do you need to be a Data Wrangler?

> Data Wranglers should have experience with programming (no specific language required, but there is a wider adoption of both ‘R’ and ‘Python’), database querying ('SQL query') and data analysis.

Reviewer: back ticks instead of single quotes for `R`, `Python`, etc.

Author: Thanks, will change

* Insufficient resources (human, computational, economic)
* Unclear scope of responsibilities within the project

In an ideal situation, some of these challenges can be mitigated if communication with Data Wranglers near the start of a project is encouraged and facilitated.  

(cl-infrastructure-data-wranglers-benefits)=
## Benefits of having Data Wranglers

* Improved communication between stakeholders: Data Wranglers will mediate the ‘language barriers’ between the different people involved in collecting the data, analysing the data and interpreting the data.
* Centrally coordinated efforts: Data Wranglers can collect common questions and discussion topics of interest and provide answers and resources.
* Data Wranglers will provide expertise in data collection, structuring and quality assurance.
This can decrease the time it takes to go from data provider to data analysis, whilst also improving analysis quality and focus.
* Data Wranglers can ensure a project keeps up to date with field specific knowledge and standards, relevant to data processing and sharing, and communicate this back to the project team.

> 

Reviewer: ```suggestion
* By participating and contributing to data analysis meetings and tasks, Data Wranglers can provide the best suitable data sets for the analysis, help understand and overcome challenges, and provide further research paths
```

Author: Will add 👍

(cl-infrastructure-data-wranglers)=
# Data Wranglers: Overview

Data Wranglers can be viewed as a specialised type of Data Scientist, primarily working in the space between data generators and data analysts.

> The terminology used to describe various data science activities (for example data collection, wrangling, analysis, modelling, visualisation), and how they map onto different job titles, is domain specific and will vary on a project and organisational level.

Reviewer 2: This sentence is very long and difficult to read, but I'm also not sure how to rewrite it.. maybe split out the two parts about the terminology and how it maps onto different job titles?

Author: Yes it is! How about - 
> "Data Wranglers can be viewed as a specialised type of Data Scientist, primarily working in the space between data generators and data analysts. There are many activities that a data scientist might undertake, for example, data collection, wrangling, analysis, modelling, visualisation and communication. How these activities map onto different job titles is domain specific and will vary on a project and organisational level."

(cl-infrastructure-data-wranglers-tasks)=
## What do Data Wranglers do?

In a data science project, it is commonly observed that data wrangling tasks take the majority of time {cite:ps}`anaconda2020sods`, in contrast to data analysis and modelling.

> Traditionally, data wrangling tasks involve cleaning, restructuring and filtering data into standardised research ready formats.

Reviewer 2: ```suggestion
Traditionally, data wrangling tasks involve cleaning, restructuring and filtering data into standardised formats.
```
That or delete the standardised part and say ready for analysis?

Reviewer 3: I like the focus on ready for analysis - data wranglers can also work in industry, not only in research

Author: Good point. We have definitely written this page with a bit of a research focus, but good to make the language more generalisable if possible

Resolved in [this commit](https://github.com/alan-turing-institute/the-turing-way/pull/2883/commits/11ad40708eb926bcaa325de61dccbc236d098160)

However, in terms of Data Wrangler as a profession, day to day tasks and objectives can be much more diverse.   

Data Wranglers work primarily in the space between data generators and data analysts, who are addressing the research question of interest.

> Understanding the intended use for the data in the context of the analysis and research questions, Data Wranglers can be in the position of influencing data generators in improving data collection methods, ultimately benefiting all points of the data life cycle.

Reviewer 2: ```suggestion
Understanding the intended use for the data in the context of the analysis and research questions, Data Wranglers can be in the position of influencing data generators in improving data collection methods. 
This ultimately benefits all points of the data life cycle.
```
How does it benefit the data life cycle? Will the data be more reusable? Will the workflow be more efficient?

Author: I am actually going to delete the the _",ultimately benefiting all points of the data life cycle"_ phase as I think it  stops the flow of this paragraph and the point should be evident from what else we say in this paragraph 🤞

Author: See this commit [here](https://github.com/alan-turing-institute/the-turing-way/pull/2883/commits/ada9c65901259f90300073bc2add049ceff695f0)

A key focus of a Data Wrangler’s role is the preparation of research ready data {cite:ps}`stewart2022table` in which data security, data management and FAIR standards {cite:ps}`mons2017fair` are all core priorities.

### Examples of day-to-day tasks:

* Data retrieval, querying, data analysis and visualizations
* Data pipeline development, maintenance and improvement
* Communicating with domain and technical experts
* Data privacy and infrastructure setup and management
* Summarising domain specific research papers
* Sharing and reviewing resources and code
* Maintaining communication across multiple teams
* Running workshops on data wrangling and related skills

(cl-infrastructure-data-wranglers-skills)=
## What qualifications or skills do you need to be a Data Wrangler?

Data Wranglers should have experience with programming (no specific language required, but there is a wider adoption of both `R` and `Python`), database querying (`SQL`) and data analysis.

> They will have an educational background to equip them to engage with the specific research data objects relevant to the projects they will work on; they will therefore have undergraduate and postgraduate degrees, or equivalent experience, giving a good grounding in these areas of study.

Reviewer 2: ```suggestion
They will have an educational background to equip them to engage with the specific research data objects relevant to the projects they will work on; they will therefore have undergraduate and postgraduate degrees, or equivalent experience.
```

Reviewer 3: I would remove the ; separating the sentences and add a full stop, and then continue with: "Therefore... "

Author: @likeajumprope addressed your comment in this commit [split sentence, JB review ](https://github.com/alan-turing-institute/the-turing-way/pull/2883/commits/a74813080d9d39a41cf6e68964d8a5e2becc3933)

> As with many data science and research infrastructure roles, further relevant training and specialisation is understood to happen on the job.

Reviewer 2: ```suggestion
As with many data science and research infrastructure roles, further relevant training and specialisation happens on the job.
```

> Additionally, a Data Wrangler will need to have great communication, problem solving and interpersonal skills, a curiosity and willingness to learn.

Reviewer 2: ```suggestion
Additionally, a Data Wrangler will need to have communication, problem solving and interpersonal skills, a curiosity and willingness to learn.
```

Reviewer 2: I'd probably leave out the communication skills out here since you touch upon those in the next sentence.

Author: Rephrased as 

> "They need good problem solving skills, with a curiosity and willingness to learn.
> Lastly, good interpersonal skills are required in order to work with people with many different backgrounds, skillsets and priorities."

They will have the ability to communicate with people with many different backgrounds, skillsets and priorities.

(cl-infrastructure-data-wranglers-challenges)=
## Challenges for Data Wranglers

> Some key challenges of a Data Wrangler role, as it is defined on this page, are:

Reviewer 2: ```suggestion
Some key challenges of a Data Wrangler role are:
```

* Stringent data privacy requirements
* Difficulties in accessing datasets
* Lack of or missing documentation on data generation, data structure and contextual information
* Insufficient resources (human, computational, economic)
* Unclear scope of responsibilities within the project

In an ideal situation, some of these challenges can be mitigated if communication with Data Wranglers near the start of a project is encouraged and facilitated.  

(cl-infrastructure-data-wranglers-benefits)=
## Benefits of having Data Wranglers

* Improved communication between stakeholders: Data Wranglers will mediate the ‘language barriers’ between the different people involved in collecting the data, analysing the data and interpreting the data.
* Centrally coordinated efforts: Data Wranglers can collect common questions and discussion topics of interest and provide answers and resources.
* Data Wranglers will provide expertise in data collection, structuring and quality assurance.
This can decrease the time it takes to go from data provider to data analysis, whilst also improving analysis quality and focus.

> * Data Wranglers can ensure a project keeps up to date with field specific knowledge and standards, relevant to data processing and sharing, and communicate this back to the project team.

Reviewer 2: ```suggestion
* Data Wranglers can ensure a project keeps up to date with field specific knowledge and standards relevant to data processing and sharing, and communicate this back to the project team.
```
